src/python/impactx/extensions/KnownElementsList.py

```diff
@@ -6,14 +6,95 @@
 License: BSD-3-Clause-LBNL
 """
 
+import os
+
+from impactx import elements
+
+
+def load_file(self, filename, nslice=1):
+    """Load and append a lattice file from MAD-X (.madx) or PALS (e.g., .pals.yaml) formats."""
+
+    # Attempt to strip two levels of file extensions to determine the schema.
+    #   Examples: fodo.madx, fodo.pals.yaml, fodo.pals.json, ...
+    file_noext, extension = os.path.splitext(filename)
+    file_noext_noext, extension_inner = os.path.splitext(file_noext)
+
+    if extension == ".madx":
+        # example: fodo.madx
+        from ..madx_to_impactx import read_lattice
+
+        self.extend(read_lattice(filename, nslice))
+        return
+
+    elif extension_inner == ".pals":
+        from pals_schema.Line import Line
+
+        # example: fodo.pals.yaml
+        if extension == ".json":
+            import json
+
+            # Read the JSON data from the test file
+            with open(filename, "r") as file:
+                json_data = json.loads(file.read())
+            # Parse the JSON data back into a Line object
+            self.from_pals(Line(**json_data), nslice)
```

Comment:
Maybe we could add an inline comment that clarifies that validation happens here when we load the data in a `Line`:
```suggestion
            # Parse the JSON data back into a Line object
            # (data validation happens here automatically)
            self.from_pals(Line(**json_data), nslice)
```